packages/vendure-plugin-shipping-extensions/src/distance-based-shipping-calculator.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  LanguageCode,
+  ShippingCalculator,
+  Injector,
+  InternalServerError,
+} from '@vendure/core';
+import { ShippingExtensionsOptions } from './shipping-extensions.plugin';
+import { PLUGIN_OPTIONS } from './constants';
+import { getDistanceBetweenPointsInKMs } from './get-distance-between-points';
+
+let pluginOptions: ShippingExtensionsOptions;
+export const distanceBasedShippingCalculator = new ShippingCalculator({
+  code: 'distance-based-shipping-calculator',
+  description: [
+    {
+      languageCode: LanguageCode.en,
+      value: 'Distance Based Shipping Calculator',
+    },
+  ],
+  args: {
+    storeLatitude: {
+      type: 'float',
+      ui: { component: 'number-form-input', min: -90, max: 90 },
+      label: [{ languageCode: LanguageCode.en, value: 'Store Latitude' }],
+    },
+    storeLongitude: {
+      type: 'float',
+      ui: { component: 'number-form-input', min: -180, max: 180 },
+      label: [{ languageCode: LanguageCode.en, value: 'Store Longitude' }],
+    },
+    pricePerKm: {
+      type: 'float',
```

Comment:
Prices are `int` in Vendure :-) The currency form input handles the floating point I think